### PR TITLE
EAB-990: add custom submit page

### PR DESCRIPTION
**Pull request by mark-roberts-ho**
Allows the use of a custom submit page

### CODE REVIEW

File: client/src/pages/forms/FormPageSubmission.jsx [hunk 1]
```diff
@@ -13,37 +42,27 @@ const FormPageSubmission = ({ businessKey }) => {
     return null;
   }
 
-  const { confirmationEmail, confirmationMessage } = formPanelContext;
+  const { confirmationEmail, confirmationMessage, confirm } = formPanelContext;
+
+  const title = confirm?.title ?? t('pages.form.cop-confirmation-panel.title');
+
+  const body = generateConfirmationMessage(
+    confirm?.message ?? confirmationMessage,
```
Comment (gardnersam) [inline client/src/pages/forms/FormPageSubmission.jsx:50]: is the confirm.message prop kind of a duplicate of the confirmationMessage one? Do we need both?

File: client/src/pages/forms/FormPageSubmission.jsx [hunk 2]
```diff
@@ -4,6 +4,35 @@ import parse from 'html-react-parser';
 import { useContext } from 'react';
 import { FormPanelContext } from '../../components/form/utils/FormPanelContext';
 
+const generateConfirmationMessage = (message, email) => {
+  const { t } = useTranslation();
+
+  return message ? (
```
Comment (gardnersam) [inline client/src/pages/forms/FormPageSubmission.jsx:10]: would it be worth adding to FormPageSubmisssion.test.jsx to cover this extra functionality? Looks like it's relatively easy to get at this code there

File: client/src/pages/forms/FormPageSubmission.jsx (hunk 1 above)
Comment (gardnersam) [reply client/src/pages/forms/FormPageSubmission.jsx:50]: If we didnt could we just extract ` confirmationTitle: form?.confirmationTitle,`

**Review by gardnersam — COMMENTED** [review]
Few thoughts / questions, looks great overall

File: client/src/pages/forms/FormPageSubmission.jsx (hunk 1 above)
Comment (mark-roberts-ho) [reply client/src/pages/forms/FormPageSubmission.jsx:50]: The confirm block is a replacement confirmationMessage prop but the latter has been retained for back compatibility

File: client/src/pages/forms/FormPageSubmission.jsx (hunk 2 above)
Comment (mark-roberts-ho) [reply client/src/pages/forms/FormPageSubmission.jsx:10]: Okat

Comment (gardnersam) [reply client/src/pages/forms/FormPageSubmission.jsx:10]: Fair enough, looks like the existing tests will cover rendering a the message input here anyway

Comment (gardnersam) [reply client/src/pages/forms/FormPageSubmission.jsx:10]: If we needed to boost test coverage later we could probably more of less copy paste the `'renders a custom confirmation message AND the email message when passed confirmationMessage AND confirmationEmail props' test and set confirmationMessage to an array, I think that would be more or less 100% coverage. I'll approve for now anyway

**Review by gardnersam — APPROVED** [review]
Looks good, thanks for the answers to my questions

**Pull request merged**